legacy.go

```diff
@@ -118,10 +118,26 @@ func (r *legacyLayerReader) walkUntilCancelled() error {
 	}
 
 	err = filepath.Walk(r.root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			// https://github.com/moby/moby/issues/32838#issuecomment-343610048
+			// Handle failure from what maybe a golang bug in the conversion of
+			// UTF16 to UTF8 in files which are left in the recycle bin. Os.Lstat
+			// which is called by filepath.Walk will fail when a filename contains
+			// unicode characters.
+			if strings.Contains(err.Error(), syscall.Errno(syscall.ERROR_FILE_NOT_FOUND).Error()) &&
```

Comment:
Can we just skip walking Files\\$Recycle.Bin altogether? I can't see why a user would ever want to commit the recycle bin directory.

Reply:
Specifically I think you just want to return filepath.SkipDir for that directory so that it doesn't even try to enumerate those files.

Reply:
I didn't know that existed. Will take a look, would make a simpler fix certainly.